xbmc/cdrip/EncoderFFmpeg.h

```diff
@@ -33,35 +33,48 @@ class CEncoderFFmpeg : public CEncoder
   bool Close() override;
 
 private:
+  struct EncoderException : public std::exception
+  {
+    std::string s;
+    EncoderException(std::string ss) : s(ss) {}
+    ~EncoderException() throw() {} // Updated
+    const char* what() const throw() { return s.c_str(); }
+  };
+
   static int avio_write_callback(void* opaque, uint8_t* buf, int buf_size);
   static int64_t avio_seek_callback(void* opaque, int64_t offset, int whence);
+
   void SetTag(const std::string& tag, const std::string& value);
   bool WriteFrame();
+  AVSampleFormat GetInputFormat(int inBitsPerSample);
+  std::string FFmpegErrorToString(int err);
 
-  AVFormatContext* m_Format{nullptr};
-  AVCodecContext* m_CodecCtx{nullptr};
-  SwrContext* m_SwrCtx{nullptr};
-  AVStream* m_Stream{nullptr};
-  AVSampleFormat m_InFormat;
-  AVSampleFormat m_OutFormat;
+  AVFormatContext* m_formatCtx{nullptr};
+  AVCodecContext* m_codecCtx{nullptr};
+  SwrContext* m_swrCtx{nullptr};
+  AVStream* m_stream{nullptr};
+  AVSampleFormat m_inFormat;
+  AVSampleFormat m_outFormat;
 
   /* From libavformat/avio.h:
    * The buffer size is very important for performance.
    * For protocols with fixed blocksize it should be set to this
    * blocksize.
    * For others a typical size is a cache page, e.g. 4kb.
    */
-  unsigned char m_BCBuffer[4096];
+  static constexpr size_t BUFFER_SIZE = 4096;
+  uint8_t* m_bcBuffer{nullptr};
 
-  unsigned int m_NeededFrames;
-  size_t m_NeededBytes;
-  uint8_t* m_Buffer{nullptr};
-  size_t m_BufferSize{0};
-  AVFrame* m_BufferFrame{nullptr};
-  uint8_t* m_ResampledBuffer{nullptr};
-  size_t m_ResampledBufferSize{0};
-  AVFrame* m_ResampledFrame{nullptr};
-  bool m_NeedConversion{false};
+  unsigned int m_neededFrames;
```

Comment:
This variable (and below) looks like it's uninitialized by the default constructor in this class.

Reply:
Will add a 0 set, normally they should not used as done on "Init" and the other using places not called if them fails. but to have set is more correct :+1:.